gobblin-data-management/src/main/java/gobblin/data/management/copy/CopyConfiguration.java

```diff
@@ -12,29 +12,44 @@
 
 package gobblin.data.management.copy;
 
+import lombok.AllArgsConstructor;
 import lombok.Data;
 
 import org.apache.hadoop.fs.Path;
 
+import com.google.common.base.Optional;
+
 
 /**
  * Configuration for Gobblin distcp jobs.
  */
 @Data
+@AllArgsConstructor
 public class CopyConfiguration {
```

Comment:
Something like `CopyConfiguration.builder(Properties props).otherMethods.build()`, the builder will parse defaults from the Properties passed in (like preserve flags, target group).